lib/sinon/assert.js

```diff
@@ -15,8 +15,11 @@ var forEach = arrayProto.forEach;
 var join = arrayProto.join;
 var splice = arrayProto.splice;
 
-function createAssertObject() {
+var DEFAULT_LOG_LIMIT = 10000;
+
+function createAssertObject(shouldLimitAssertionLogs) {
     var assert;
+    var assertionLogLimit;
```

Comment:
```suggestion
```
This only makes sense to keep if you expose a way of setting it. So in the current context, it could be removed, but I assume it has been added to allow for testing this, which would 

1. Fix the only remaining issue
2. Make this line of code meaningful 😄 

P.S. We [allow newer EcmaScript syntax](https://github.com/sinonjs/sinon/blob/main/COMPATIBILITY.md#es2017), so you mist as well use `const`.